api/src/main/java/marquez/db/DbRetention.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018-2023 contributors to the Marquez project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package marquez.db;
+
+import com.google.common.base.Stopwatch;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import marquez.db.exceptions.DbRetentionException;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.StatementException;
+
+@Slf4j
+public final class DbRetention {
+  private DbRetention() {}
+
+  /* Bulk delete batch size. */
+  private static final int BULK_DELETE_BATCH_SIZE = 1000;
```

Comment:
Should we expose this in the CLI as well, so users can control the `LIMIT`?
Was this default value set based on some empirical value?

Reply:
Also, this should have an explanation in the comment on why we need to limit this.

Reply:
I made the chunk size configurable for the CLI and retention job. For example:

```yaml
dbRetention:
  frequencyMins: ${DB_RETENTION_FREQUENCY_MINS:-15}
  # Maximum data chunk size that can be deleted per retention execution (default: 1000)
  chunkSize:  ${DB_RETENTION_CHUNK_SIZE:-1000}
  retentionDays: ${DB_RETENTION_DAYS:-7}
```